### ISSUE TITLE
Train M1 v4 with expanded semantic categories

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#499]
## Context

After importing and merging external semantic categories (#493), we need to train M1 v4 with the expanded data.

**Related issues:**
- #490: Import external semantic categories (parent)
- #493: Merge external categories and validate coverage (prerequisite)
- #500: Measure M1 v4 quality improvements (validation)

## Current State (M1 v3)

**Training data stats:**
- 1,220 categorized nouns (69% coverage)
- 3.2% semantic violation rate
- 32.3% constraint triple object coverage

**Model performance:**
- Test accuracy: ~75-80%
- Practical validation: 4/6 correct (66.7%)

## Expected State (M1 v4)

**Training data stats (after #493):**
- 1,400+ categorized nouns (80%+ coverage)
- 40-50% semantic violation rate
- 60%+ constraint triple object coverage

**Model performance goals:**
- Test accuracy: >85%
- Practical validation: 5/6 or 6/6 correct (83-100%)

### Step 1: Regenerate Training Data

```bash
# This uses semantic_categories_merged.json by default (updated in #493)
./scripts/m1_generate_semantic_data.sh --fresh
```

**Verification:**
```bash
# Check training data stats
python -c "
import json
from pathlib import Path

stats_file = Path('data/training/m1_semantic_violations/stats.json')
if stats_file.exists():
    stats = json.load(open(stats_file))
    print(f'Semantic violation rate: {stats[\"semantic_violation_rate\"]*100:.1f}%')
    print(f'Constraint coverage: {stats[\"constraint_coverage\"]*100:.1f}%')
"
```

### Step 2: Train M1 Selectional Model

```bash
./scripts/m1_train_selectional.sh --fresh
```

**Expected output:**
- Checkpoint: `models/m1_selectional/best_model.pt`
- Training log: `models/m1_selectional/training.log`
- Final test accuracy printed

### Step 3: Validate Model

```bash
./scripts/m1_validate_selectional.sh
```

**Expected output:**
- Test accuracy score
- Per-category metrics
- Example predictions

## Dependencies

**Depends on:**
- #493: Merge external categories (must complete first)
- `semantic_categories_merged.json` must exist

**Blocks:**
- #500: Quality measurement (needs trained model)

**Optional enhancements:**
- #497: Retry logic (makes import more robust, but not required)
- #498: Category mapper expansion (improves coverage, but not required)

## Timeline

**When to do this:**
- Immediately after #493 completes
- Before measuring quality improvements (#500)

**Estimated time:**
- Data generation: 5-10 minutes
- Training: 15-30 minutes (depending on hardware)
- Validation: 2-3 minutes

## Priority

P1 (high priority)
- This is the goal of #490 parent issue
- Can't validate improvements without trained model
- Relatively quick to execute

## Acceptance Criteria

- [ ] Training data regenerated with merged categories
- [ ] Semantic violation rate ≥40% (up from 3.2%)
- [ ] M1 model trained successfully
- [ ] Checkpoint saved to `models/m1_selectional/best_model.pt`
- [ ] Test accuracy measured and logged
- [ ] Ready for quality validation (#500)

## Success Metrics

**Training data:**
- ✓ Semantic violation rate: 3.2% → 40%+
- ✓ Constraint coverage: 32.3% → 60%+

**Model performance:**
- ✓ Test accuracy: 75-80% → 85%+
- ✓ Practical validation: 4/6 → 5/6 or 6/6

## Notes

- Training scripts already support `--fresh` flag (implemented in previous work)
- Shell wrappers handle venv activation automatically
- Checkpoint rotation keeps last 2 models for safety
- Early stopping prevents overfitting (patience=3 epochs)

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
